exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.stats.stackdriver;
+
+import com.google.api.MetricDescriptor;
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.monitoring.v3.CreateMetricDescriptorRequest;
+import com.google.monitoring.v3.ProjectName;
+import io.opencensus.exporter.metrics.util.MetricExporter;
+import io.opencensus.metrics.export.Metric;
+import io.opencensus.metrics.export.MetricDescriptor.Type;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CreateMetricDescriptorExporter extends MetricExporter {
+  private static final Tracer tracer = Tracing.getTracer();
+  private static final Logger logger =
+      Logger.getLogger(CreateMetricDescriptorExporter.class.getName());
+
+  private final String projectId;
+  private final ProjectName projectName;
+  private final MetricServiceClient metricServiceClient;
+  private final String domain;
+  private final String displayNamePrefix;
+  private final Map<String, io.opencensus.metrics.export.MetricDescriptor>
+      registeredMetricDescriptors = new LinkedHashMap<>();
+  private final MetricExporter nextExporter;
```

Comment:
nit: the type of `nextExporter` can be restricted to `CreateTimeSeriesExporter`.

Reply:
I use a different MetricExporter in tests. Also this class is internal.